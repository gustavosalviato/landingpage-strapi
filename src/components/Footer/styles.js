import styled, { css } from "styled-components"
import { Text } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
            text-align: center;
            border-top: 0.1rem solid #DDD;

            a{
                text-decoration: none;
            }            

            ${Text}{
                font-size: ${theme.font.sizes.small};
            }      

           
    `}
`







