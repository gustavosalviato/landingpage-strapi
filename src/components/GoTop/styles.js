import styled, { css } from "styled-components";

export const Container = styled.a`
    ${({ theme, background }) => css`
        position: fixed;
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: ${theme.colors.darkBg};
        color: ${theme.colors.lightBg};
        width: 4rem;
        height: 4rem;
        bottom: 4rem;
        right: 4rem;
        z-index: 6;
        `}
`