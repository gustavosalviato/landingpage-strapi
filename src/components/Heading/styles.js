import styled, { css } from 'styled-components'

export const titleSize = {
    small: (theme) => css`
        font-size: ${theme.font.sizes.small}
    `,

    medium: (theme) => css`
        font-size: ${theme.font.sizes.medium}
    `,

    large: (theme) => css`
        font-size: ${theme.font.sizes.large}
    `,
};

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;



export const Title = styled.h1`
    ${({ theme, size, uppercase }) => css`
        color: ${theme.colors.lightBg};
        ${titleSize[size](theme)};
        ${titleCase(uppercase)}
    `}; 

    

`;