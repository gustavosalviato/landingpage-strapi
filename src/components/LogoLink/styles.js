import styled, { css } from "styled-components";


export const Container = styled.a`
    ${({ theme }) => css`
        text-decoration: none;
        color: inherit;

        >img{
            height: 6rem;
        }
    `}
`