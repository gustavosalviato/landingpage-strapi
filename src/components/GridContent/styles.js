import styled, { css } from "styled-components";
import { Title } from '../Heading/styles';
import { Text } from '../TextComponent/styles';
export const Container = styled.div`
    ${({ theme, background }) => css`
        max-width: 58rem;
        text-align: center;
        margin: 0 auto;

        ${Title}{
            color: ${background ? theme.colors.darkBg : theme.colors.lightBg};
        }
    `}
`

export const Html = styled.div`
    ${({ theme, background }) => css`
        margin: ${theme.spacings.xlarge} 0;

        ${Text}{
        color: ${background ? theme.colors.darkBg : theme.colors.lightBg};        
    }
    `}

   
`
