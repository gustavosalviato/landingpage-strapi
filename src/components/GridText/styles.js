import styled, { css } from 'styled-components'
import { Title } from '../Heading/styles'
import { Text } from '../TextComponent/styles'
export const Container = styled.div`
    ${({ theme }) => css`
        ${Text}{
            margin-bottom: ${theme.spacings.small};;
        }

    `}
`
export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.medium};
        overflow: hidden;
        width: 100%;

        @media ${theme.media.lteMedium}{
            grid-template-columns: 1fr;
            
        }
    `}
`

export const GridElement = styled.div`
    ${({ theme }) => css`
        ${Title}{
            position: relative;
            left: 5rem
        }
    

        ${Title}::before{
            counter-increment: grid-counter;
            content: counter(grid-counter);
            position: absolute;
            left: -5rem;
            font-size: 6rem;
            top: -2rem;
            transform: rotate(10deg);
        }
    
    `}
`



