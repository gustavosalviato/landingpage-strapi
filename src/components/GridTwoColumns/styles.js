import styled, { css } from "styled-components";
import { Title } from '../Heading/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacings.large};

        @media ${theme.media.lteMedium}{
            grid-template-columns: 1fr;
            text-align: center;
        }

        ${Title}{
            margin-bottom: ${theme.spacings.large}
        }
    `}

`

export const TextContainer = styled.div`
    ${({ theme }) => css`
        @media ${theme.media.lteMedium}{
            margin-bottom: ${theme.spacings.large};
        }
        
    `}

`


export const ImageContaner = styled.div`
    ${({ theme }) => css`
        width: 600px;
        height: 600px;
    `}

`


export const Image = styled.img`
    ${({ theme }) => css`
        width: 100%;
        fill: red;
    `}

`