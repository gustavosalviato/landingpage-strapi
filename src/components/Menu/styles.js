import styled, { css } from "styled-components";
import { Container as SectionContainer } from '../SectionContainer/styles'
import { Title as Heading } from '../Heading/styles';

const menuVisible = (theme) => css`
    visibility: visible;
    opacity: 1;
`;

//Style for Mobile
export const Container = styled.div`
 ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    bottom: 0;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    background: ${theme.colors.darkBg};
    transition: all 300ms ease-in-out;
 
    >${SectionContainer}{
        padding-top: 0;
        padding-bottom: 0;
    }


    & ${Heading}{
        padding-top: 0;
        padding-bottom: 0;
    }

    @media ${theme.media.lteMedium}{
    height: 100vh;
    visibility: hidden;
    opacity: 0;
    ${visible && menuVisible(theme)}

    > ${SectionContainer}{
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        overflow-y: auto;
        scroll-behavior: smooth;

    }


    & ${Heading}{
        display: flex;
        justify-content: center;
        padding-bottom: ${theme.spacings.large};
    }

 }`}
 
`;

export const MenuContainer = styled.div`
    ${({ theme }) => css`
        display: flex;
        justify-content: space-between;
        align-items: center;


        @media ${theme.media.lteMedium} {
        display: block;
        text-align: center;
        padding: ${theme.spacings.large} 0;
    }
    `}


`;

export const Button = styled.button`
    ${({ theme, visible }) => css`
        z-index: 6;
        position: fixed;
        top: 2rem;
        right: 2rem;
        width: 4rem;
        height: 4rem;
        background: ${theme.colors.darkBg};
        color: ${theme.colors.lightBg};
        border: none;
        display: none;
        cursor: pointer;
        pointer-events: ${visible ? 'none' : 'all'};

        @media ${theme.media.lteMedium}{
            display: flex;
            justify-content: center;
            align-items: center;
        }
    `}

    >svg{
        width: 2.5rem;
        height: 2.5rem;
    }


`;
