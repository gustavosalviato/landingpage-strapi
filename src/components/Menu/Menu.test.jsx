import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from "../../styles/render-theme";
import { Menu } from '.'
import Links from '../NavLink/mock'
import { theme } from '../../styles/theme'
const logoData = {
    text: 'Logo',
    link: '#target',
    src: 'image.png'
}

describe('<Menu />', () => {
    it('should render Logo and  MainMenu', () => {
        const { container } = renderTheme(<Menu links={Links} logoData={logoData} />);
        expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
        expect(screen.getByRole('navigation', { name: 'Main Menu' })).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });



    it('should render menu mobile and button for open and close the menu', () => {
        const { container } = renderTheme(<Menu links={Links} logoData={logoData} />);
        const button = screen.getByLabelText('Open/Close Menu');
        // const menuButton = button.nextSibling();
        expect(button).toHaveStyle('display: none');

        expect(screen.getByLabelText('Open Menu')).toBeInTheDocument();

        fireEvent.click(button);

        expect(screen.getByLabelText('Close Menu')).toBeInTheDocument();
    });

    it('should not render links', () => {
        const { container } = renderTheme(<Menu logoData={logoData} />);

        expect(screen.queryByRole('navigation', { name: 'Main Menu' }).firstChild).not.toBeInTheDocument();

        expect(container).toMatchSnapshot();
    })

});







