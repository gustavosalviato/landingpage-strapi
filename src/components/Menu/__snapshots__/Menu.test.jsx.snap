// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should not render links 1`] = `
<div>
  <button
    aria-label="Open/Close Menu"
    class="sc-eCYdqJ htFPjf"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="sc-dkzDqf iAYlfV"
  >
    <div
      class="sc-bczRLJ bFmWNK"
    >
      <div
        class="sc-hKMtZM jfsuhj"
      >
        <h1
          class="sc-gsnTZi hQyYne"
        >
          <a
            class="sc-jSMfEi iZjaRP"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main Menu"
          class="sc-gKXOVf hjDHrt"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> should render Logo and  MainMenu 1`] = `
<div>
  <button
    aria-label="Open/Close Menu"
    class="sc-eCYdqJ htFPjf"
  >
    <svg
      aria-hidden="true"
      aria-label="Open Menu"
      class="StyledIconBase-ea9ulj-0 lbJwfL"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0V0z"
        fill="none"
      />
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="sc-dkzDqf iAYlfV"
  >
    <div
      class="sc-bczRLJ bFmWNK"
    >
      <div
        class="sc-hKMtZM jfsuhj"
      >
        <h1
          class="sc-gsnTZi hQyYne"
        >
          <a
            class="sc-jSMfEi iZjaRP"
            href="#target"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main Menu"
          class="sc-gKXOVf hjDHrt"
        >
          <a
            class="sc-iBkjds cjQOBn"
            href="#target1"
            target="_self"
          >
            Link 1
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target2"
            target="_self"
          >
            Link 2
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target3"
            target="_self"
          >
            Link 3
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target4"
            target="_self"
          >
            Link 4
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target5"
            target="_self"
          >
            Link 5
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target6"
            target="_self"
          >
            Link 6
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target7"
            target="_self"
          >
            Link 7
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target8"
            target="_self"
          >
            Link 8
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target9"
            target="_self"
          >
            Link 9
          </a>
          <a
            class="sc-iBkjds cjQOBn"
            href="#target10"
            target="_self"
          >
            Link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
